### PR TITLE
[T5A2][F12-A3]Joel Lim Jing

### PR DESCRIPTION
**Pull request by aljorhythm**

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -65,4 +65,9 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Returns true if command mutates data
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:70]: the command can *potentially* mutate data. may be you can update the comment to reflect that.
e.g. add command doesn't mutate data if the user tries to add a duplicate entry.

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +84,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java:87]: you need to update the documentation and the tests to reflect the changed behavior of this method

**Review by okkhoy — COMMENTED** [review]
Notice that your tests fail! this is because, you haven't updated the tests to check for the changed behavior of execute command.

**Comment by okkhoy** [comment]
@funnieguy 
Some comments added. Please ack & close the PR after reading comments.

**Comment by aljorhythm** [comment]
Ack

**Pull request closed without merging**